src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,145 @@
+        String name = "";
+        if (command.equals("deadline")) {
+            String deadline = "";
+            boolean completedName = false;
+            for (int i = 1; i < arrStrings.length; i++) {
+                if (arrStrings[i].equals("/by")) {
+                    completedName = true;
+                    continue;
+                }
+                if (completedName) {
+                    deadline += arrStrings[i] + " ";
+                } else {
+                    name += arrStrings[i] + " ";
+                }
+            }
+            if (!completedName) {
+                throw new Exception("Invalid deadline task!");
+            }
+            return new Deadline(name.substring(0, name.length() - 1), deadline.substring(0, deadline.length() - 1));
+        } else if (command.equals("todo")) {
+            if (arrStrings.length == 1) {
+                throw new Exception("OOPS!!! The description of a todo cannot be empty.");
+            }
+            for (int i = 1; i < arrStrings.length; i++) {
+                name += arrStrings[i] + " ";
+            }
+            return new Todo(name.substring(0, name.length() - 1));
+        } else if (command.equals("event")) {
+            String from = "";
+            String to = "";
+            boolean completedName = false;
+            boolean completedFrom = false;
+            for (int i = 1; i < arrStrings.length; i++) {
+                if (arrStrings[i].equals("/from")) {
+                    completedName = true;
+                } else if (arrStrings[i].equals("/to")) {
+                    completedFrom = true;
+                } else if (!completedFrom && completedName) {
+                    from += arrStrings[i] + " ";
+                } else if (completedFrom && completedName) {
+                    to += arrStrings[i] + " ";
+                } else {
+                    name += arrStrings[i] + " ";
+                }
+            }
+            return new Event(name.substring(0, name.length() - 1), from.substring(0, from.length() - 1),
+                    to.substring(0, to.length() - 1));
+        } else {
+            throw new Exception("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    /**
+     * Parse the input string by splitting based on whtiespaces.
+     * Sorts the string based on its command
+     *
+     * @param input The input string representing the task given by user
+     * @param ui The user interface responsible for sending outputs to the screen
+     * @param taskList ArrayList that stores the tasks loaded from memory
+     * @param storage Storage that can be written to or read from
+     */
+    public static String parse(String input, Ui ui, TaskList taskList, Storage storage) {
+        // Splits the input based on whitespaces.
+        String command = input.split("\\s+")[0];
+        String message = "";
+        int choice = -1;
+        switch (command) {
```

Comment:
Good that you have followed the course's recommended switch case indentation level!